### PR TITLE
First recipe activity

### PR DESCRIPTION
**Pull request by asos-emma-burton**
Shows list of recipes (not yet clickable), with images and names

image

### CODE REVIEW

File: app/src/main/java/com/example/emmaburton/projectemma/data/model/Recipe.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        @SerializedName("steps") val steps: Array<String>,
+
+        @SerializedName("timers") val timers: Array<Int>,
+
+        @SerializedName("imageURL") val imageURL: String,
+
+        @SerializedName("originalURL") val originalURL: String
+
+) {
+    override fun equals(other: Any?): Boolean {
```
Comment (asos-olivier) [inline app/src/main/java/com/example/emmaburton/projectemma/data/model/Recipe.kt]: kotlin data classes are already implementing equals and hashcode for you so no need to do it! :-)

File: app/src/main/java/com/example/emmaburton/projectemma/presentation/activity/RecipeListActivity.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private val api = RecipesApi()
+    private val compositeDisposable = CompositeDisposable()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        this.setContentView(R.layout.activity_recipe_list)
+
+        search_results_recycler_view.layoutManager = GridLayoutManager(this, 1)
+
+        compositeDisposable.add(api.getAllRecipes()
```
Comment (asos-olivier) [inline app/src/main/java/com/example/emmaburton/projectemma/presentation/activity/RecipeListActivity.kt]: If you want a nice testable app, you will need to move this call to another class (a Presenter or a ViewModel) as doing it here in the UI will make it harder to test and maintain.

File: app/src/main/java/com/example/emmaburton/projectemma/data/model/Recipe.kt (hunk 1 above)
Comment (asos-emma-burton) [reply app/src/main/java/com/example/emmaburton/projectemma/data/model/Recipe.kt]: but i saw this warning? should I ignore it? 😬 
image

**Comment by asos-chamarro** [comment]
It would be nice to modify the `.gitignore` file to keep the `/build` folder out from the repository. You could copy the ASOS one or this one (https://github.com/xaman/Muvis/blob/master/.gitignore)

**Pull request closed without merging**